### PR TITLE
WIP: Fix incorrect test vectors.

### PR DESCRIPTION
**Pull request by nochiel**
Fixes for <https://github.com/BlockchainCommons/Research/issues/83>.

Ref. <https://github.com/BlockchainCommons/sweeptool-cli/issues/1>

- papers\bcr-2020-010-output-desc.md
    - [x] Test vector 4.
    - ~[x] Test vector 5.~
      - No changes needed.

### CODE REVIEW

**Comment by nochiel** [comment]
RE; <https://github.com/BlockchainCommons/sweeptool-cli/issues/1#issuecomment-879802335>
@gorazdko @wolfmcnally Added for discussion as I work on this. Do these changes make sense?

File: papers/bcr-2020-007-hdkey.md [hunk 1]
```diff
@@ -54,14 +54,14 @@ When used embedded in another CBOR structure, this structure should be tagged #6
 ; If `components` is empty, then `source-fingerprint` MUST be a fingerprint of
 ; a master key.
 ;
-; `depth`, if present, represents the number of derivation steps in
```
Comment (gorazdko) [inline papers/bcr-2020-007-hdkey.md:57]: I'd leave this as it was and change only to: regardless whether present in the components path

File: papers/bcr-2020-007-hdkey.md [hunk 2]
```diff
@@ -54,14 +54,14 @@ When used embedded in another CBOR structure, this structure should be tagged #6
...
+; `depth`, represents the number of derivation steps in
 ; the path of the associated key, even if not present in the `components` element
 ; of this structure.
 
 crypto-keypath = {
 	components: [path-component], ; If empty, source-fingerprint MUST be present
 	? source-fingerprint: uint32 .ne 0 ; fingerprint of ancestor key, or master key if components is empty
-	? depth: uint8 ; 0 if this is a public key derived directly from a master key
```
Comment (gorazdko) [inline papers/bcr-2020-007-hdkey.md]: would leave this as it was as depth is still optional most of the time

File: papers/bcr-2020-010-output-desc.md [hunk 3]
```diff
@@ -242,12 +242,14 @@ fffffffe ; child number
 			1: [ ; components
 				44, true, 0, true, 0, true ; 44'/0'/0'
 			],
-			2: 3545084735 ; origin-fingerprint
+			2: 3545084735 ; source-fingerprint
+                        3: 4 ; depth
```
Comment (gorazdko) [inline papers/bcr-2020-010-output-desc.md]: can you use tabs here as in previous row for alignment

File: papers/bcr-2020-010-output-desc.md [hunk 4]
```diff
@@ -242,12 +242,14 @@ fffffffe ; child number
...
 				44, true, 0, true, 0, true ; 44'/0'/0'
 			],
-			2: 3545084735 ; origin-fingerprint
+			2: 3545084735 ; source-fingerprint
+                        3: 4 ; depth    
 		}),
 		7: 304({ ; children: crypto-keypath
 			1: [ ; components
 				1, false, [], false ; 1/*
-			]
```
Comment (gorazdko) [inline papers/bcr-2020-010-output-desc.md]: no changes are needed here

Comment (nochiel) [reply papers/bcr-2020-010-output-desc.md]: > no changes are needed here

😅 I'm reverting that line but I don't understand why we don't need the `depth` for `children`.

File: papers/bcr-2020-010-output-desc.md [hunk 5]
```diff
@@ -288,20 +291,20 @@ D9 0193                                 # tag(403) public-key-hash
...
+D90193D9012FA503582102D2B36900396C9282FA14628566582F206A5DD0BCC8D5E892611806CAFB0301F0045820637807030D55D01F9A0CB3A7839515D796BD07706386A6EDDF06CC29A65A0E2906D90130A30186182CF500F500F5021AD34DB33F030407D90130A1018401F480F4081A78412E3A
 ```
 
 * As a UR:
 
 ```
-ur:crypto-output/taadmutaaddlonaxhdclaotdqdinaeesjzmolfzsbbidlpiyhddlcximhltirfsptlvsmohscsamsgzoaxadwtaahdcxiaksataxbtgotictnybnqdoslsmdbztsmtryatjoialnolweuramsfdtolhtbadtamtaaddyoeadlncsdwykaeykaeykaocytegtqdfhattaaddyoyadlradwklawkaycyksfpdmftpyaaeelb
+ur:crypto-output/fyesdyeheseofyesdyeheyfgfpecdyeoeceteyehdyeyfyeyfweoenesdydyeoesenfxeseyeteyfgfpeheeeneyeteceneneceteyfgeydyenfpecfyfydyfwfxfxetfyecfeeteseyenehehetdyenfxfpfgfwdyeodyehfgdydyeeeceteydyeneoemetdyemdyeodyfyececfydyehfgesfpdyfxfweofpemeteoesecehecfyemesenfwfydyememdyeneoetenfpenfefyfyfgdyenfxfxeyesfpenecfpdyfeeyesdyenfyesdyeheodyfpeodyeheteneheteyfxfgecdydyfgecdydyfgecdyeyehfpfyeoeefyfweoeofgdyeodyeedyemfyesdyeheodyfpehdyeheteedyehfgeeetdyfgeedyetehfpemeteeeheyfeeofpgyvdhfva
```
Comment (gorazdko) [inline papers/bcr-2020-010-output-desc.md]: cbor looks good. 

However for UR i get:

```
$ bytewords -i hex D90193D9012FA503582102D2B36900396C9282FA14628566582F206A5DD0BCC8D5E892611806CAFB0301F0045820637807030D55D01F9A0CB3A7839515D796BD07706386A6EDDF06CC29A65A0E2906D90130A30186182CF500F500F5021AD34DB33F030407D90130A1018401F480F4081A78412E3A  -o minimal
taadmutaaddlonaxhdclaotdqdinaeesjzmolfzsbbidlpiyhddlcximhltirfsptlvsmohscsamsgzoaxadwtaahdcxiaksataxbtgotictnybnqdoslsmdbztsmtryatjoialnolweuramsfdtolhtbadtamtaaddyotadlncsdwykaeykaeykaocytegtqdfhaxaaattaaddyoyadlradwklawkaycyksfpdmftkiiozsfd
```

Can you fix this and squash your commits?

Comment (nochiel) [reply papers/bcr-2020-010-output-desc.md]: > Can you fix this and squash your commits?

Will do. 

I was using [foundation-ur](https://github.com/Foundation-Devices/foundation-ur-py). I'm not sure why I was getting a different output. I'll update after I understand what I was doing wrong.

**Review by wolfmcnally — APPROVED** [review]
LGTM

**Pull request merged**